scipy/optimize/_linprog_ip.py

```diff
@@ -0,0 +1,1940 @@
+
+    # identical bounds indicate that variable can be removed
+    i_f = np.abs(lb - ub) < tol   # indices of "fixed" variables
+    i_nf = np.logical_not(i_f)  # indices of "not fixed" variables
+    if np.any(i_f):
+        c0 += c[i_f].dot(lb[i_f])
+        b_eq = b_eq - A_eq[:, i_f].dot(lb[i_f])
+        b_ub = b_ub - A_ub[:, i_f].dot(lb[i_f])
+        c = c[i_nf]
+        x = x[i_nf]
+        A_eq = A_eq[:, i_nf]
+        A_ub = A_ub[:, i_nf]
+        # record of variables to be added back in
+        undo = [np.where(i_f)[0], lb[i_f]]
+        # don't remove these entries from bounds; they'll be used later.
+
+    # no constraints indicates that problem is trivial
+    if A_eq.size == 0 and A_ub.size == 0:
+        b_eq = np.array([])
+        b_ub = np.array([])
+        # test_empty_constraint_1
+        if c.size == 0:
+            status = 0
+            message = "The solution was determined in presolve as there are \
+            no non-trivial constraints."
+        elif np.any(np.logical_and(c < 0, ub == np.inf)) or \
+                np.any(np.logical_and(c > 0, lb == -np.inf)):
+                # test_no_constraints()
+            status = 3
+            message = "If feasible, the problem is (trivially) unbounded " \
+                "because there are no constraints and at least one element " \
+                "of c is negative. If you wish to check whether the " \
+                "problem is infeasible, turn presolve off."
+        else:  # test_empty_constraint_2
+            status = 0
+            message = "The solution was determined in presolve as there are " \
+                "no non-trivial constraints."
+        complete = True
+        x[c < 0] = ub[c < 0]
+        x[c > 0] = lb[c > 0]
+        # if this is not the last step of presolve, should convert bounds back
+        # to array and return here
+
+    # *sigh* - convert bounds back to their standard form (list of tuples)
+    # again, in retrospect, numpy array would be standard form
+    lb[np.equal(lb, -np.inf)] = None
+    ub[np.equal(ub, np.inf)] = None
+    bounds = np.hstack((lb[:, np.newaxis], ub[:, np.newaxis]))
+    bounds = bounds.tolist()
+    for i, row in enumerate(bounds):
+        for j, col in enumerate(row):
+            if str(
+                    col) == "nan":  # comparing col to float("nan") and
+                                    # np.nan doesn't work. should use np.isnan
+                bounds[i][j] = None
+
+    # remove redundant (linearly dependent) rows from equality constraints
+    if A_eq.size > 0 and np.linalg.matrix_rank(A_eq) < A_eq.shape[0]:
+        warn("A_eq does not appear to be of full row rank. To improve "
+             "performance, check the problem formulation for redundant "
+             "equality constraints.", OptimizeWarning)
+        A_eq, b_eq, status, message = _remove_redundancy(A_eq, b_eq)
+        if status != 0:
+            complete = True
+
+    return c, c0, A_ub, b_ub, A_eq, b_eq, bounds, \
```

Comment:
There's a lot of ``\``s here. In general we try to avoid those, both because they look outdated and because putting a space behind them (not hard to do) gives ``SyntaxError``s.